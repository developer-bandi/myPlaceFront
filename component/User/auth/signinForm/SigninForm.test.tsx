import { act, renderHook } from "@testing-library/react";
import axios from "axios";
import newReact from "react";
import * as React from "react";
import { useRouter } from "next/router";
import useSigninForm from "./SigninFormHook";
const CryptoJS = require("crypto-js");

let status = {
  post: 202,
};
jest.mock("axios", () => {
  return {
    create: jest.fn().mockReturnValue({
      interceptors: {
        request: { use: jest.fn(), eject: jest.fn() },
        response: { use: jest.fn(), eject: jest.fn() },
      },
      post: jest.fn(() => {
        if (status.post === 202) {
          return Promise.resolve({
            status: 202,
            data: "비밀번호 혹은 아이디를 잘못입력하였습니다",
          });
        }
        if (status.post === 200) {
          return Promise.resolve({
            status: 200,
            data: "로그인 성공",
          });
        }
        if (status.post === 500) {
          return Promise.reject({
            status: 500,
            data: "에러 발생",
          });
        }
      }),
    }),
  };
});

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));
const mockRouter = {
  push: jest.fn(),
};
(useRouter as jest.Mock).mockReturnValue(mockRouter);

const alertMock = jest.fn();
window.alert = alertMock;

jest.spyOn(CryptoJS.AES, "encrypt").mockReturnValue({ toString: () => "test" });

describe("SigninForm Hook 테스트", () => {
  describe("checkLogin 함수 테스트", () => {
    it("잘못된 아이디 혹은 비밀번호를 입력한경우", async () => {
      jest
        .spyOn(newReact, "useRef")
        .mockReturnValue({ current: { value: "" } });

      const { result } = renderHook(() => useSigninForm());
      await act(async () => {
        await result.current.checkLogin({ type: "click" });
      });
      expect(alertMock.mock.calls[0][0]).toBe(
        "비밀번호 혹은 아이디를 잘못입력하였습니다"
      );
    });
    it("로그인에 성공한 경우", async () => {
      jest
        .spyOn(newReact, "useRef")
        .mockReturnValue({ current: { value: "" } });
      status.post = 200;
      const { result } = renderHook(() => useSigninForm());
      await act(async () => {
        await result.current.checkLogin({ type: "click" });
      });
      expect(alertMock.mock.calls[1][0]).toBe("로그인에 성공하였습니다!");
      expect(mockRouter.push).toBeCalledWith("/");
    });
    it("에러가 발생한 경우", async () => {
      jest
        .spyOn(newReact, "useRef")
        .mockReturnValue({ current: { value: "" } });
      status.post = 500;
      const { result } = renderHook(() => useSigninForm());
      await act(async () => {
        await result.current.checkLogin({ type: "click" });
      });
      expect(alertMock.mock.calls[2][0]).toBe("서버에 에러가 발생하였습니다");
    });
  });
});
