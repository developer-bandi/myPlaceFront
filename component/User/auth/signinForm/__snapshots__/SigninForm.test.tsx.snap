// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SigninForm Presentational 테스트 정상 출력 1`] = `
<div>
  <main
    class="mainBlock"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        로그인
      </h1>
      <a
        class="naverBlock"
        href="undefined/auth/naver"
        target="_blank"
      >
        <div
          class="naverContentBlock"
        >
          <div
            class="naverLogo"
          >
            N
          </div>
          <div
            class="naverText"
          >
            네이버 로그인
          </div>
        </div>
      </a>
      <a
        class="kakaoBlock"
        href="undefined/auth/kakao"
        target="_blank"
      >
        <div
          class="kakaoContentBlock"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2725%27%20height=%2725%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="kakaologo"
              class="kakaoLogo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div
            class="kakaoText"
          >
            카카오 로그인
          </div>
        </div>
      </a>
      <div
        class="divisionBlock"
      >
        <hr
          class="divisionBar"
        />
        <p
          class="divisionText"
        >
          또는
        </p>
        <hr
          class="divisionBar"
        />
      </div>
      <div
        class="contentBlock"
      >
        <div
          class="subTitle"
        >
          아이디
        </div>
        <input
          class="input"
          data-testid="id"
        />
      </div>
      <div
        class="contentBlock"
      >
        <div
          class="subTitle"
        >
          비밀번호
        </div>
        <input
          class="input"
          data-testid="password"
          type="password"
        />
      </div>
      <div
        class="submitButtonBlock"
      >
        <button
          class="submitButton"
          data-testid="checklogin"
        >
          로그인하기
        </button>
      </div>
      <ul
        class="bottomBlock"
      >
        <li
          class="bottomLink"
        >
          <a
            href="/user/auth/signup"
          >
            회원가입
          </a>
        </li>
        <li
          class="bottomLink"
        >
          <a
            href="/user/auth/findid"
          >
            아이디 찾기
          </a>
        </li>
        <li
          class="bottomLink"
        >
          <a
            href="/user/auth/findpassword"
          >
            비밀번호 찾기
          </a>
        </li>
      </ul>
    </div>
  </main>
</div>
`;
